python/cudf_polars/cudf_polars/experimental/benchmarks/pdsh.py

```diff
@@ -1153,48 +1156,68 @@ def run(args: argparse.Namespace) -> None:
                     raise ImportError from err
 
     records: defaultdict[int, list[Record]] = defaultdict(list)
+    engine: pl.GPUEngine | None = None
+
+    if run_config.executor == "cpu":
+        engine = None
+    else:
+        executor_options: dict[str, Any] = {}
+        if run_config.executor == "streaming":
+            executor_options = {
+                "cardinality_factor": {
+                    "c_custkey": 0.05,  # Q10
+                    "l_orderkey": 1.0,  # Q18
+                    "l_partkey": 0.1,  # Q20
+                    "o_custkey": 0.25,  # Q22
+                },
+            }
+            if run_config.blocksize:
+                executor_options["target_partition_size"] = run_config.blocksize
+            if run_config.shuffle:
+                executor_options["shuffle_method"] = run_config.shuffle
+            if run_config.broadcast_join_limit:
+                executor_options["broadcast_join_limit"] = (
+                    run_config.broadcast_join_limit
+                )
+            if run_config.rapidsmpf_spill:
+                executor_options["rapidsmpf_spill"] = run_config.rapidsmpf_spill
+            if run_config.scheduler == "distributed":
+                executor_options["scheduler"] = "distributed"
+
+        engine = pl.GPUEngine(
+            raise_on_fail=True,
+            executor=run_config.executor,
+            executor_options=executor_options,
+        )
+
     for q_id in run_config.queries:
         try:
             q = getattr(PDSHQueries, f"q{q_id}")(run_config)
         except AttributeError as err:
             raise NotImplementedError(f"Query {q_id} not implemented.") from err
 
+        if run_config.executor == "cpu":
+            if args.explain_logical:
+                print(f"\nQuery {q_id} - Logical plan\n")
+                print(q.explain())
```

Comment:
This is new: it prints out the polars (logical) plan with `-e cpu --explain-logical`.